src/arrayMethodFilter.js

```diff
@@ -5,7 +5,12 @@
  */
 function applyCustomFilter() {
   [].__proto__.filter2 = function(callback) {
-    // write code here
+    let result = [];
+    this.map((el, i) => {
+      return callback(el, i, this)
```

Comment:
I'd recommend using the next approach for splitting ternary to multiple lines
```
condition
  ? ... code if true
  : ... code if false
```
This way the code is more readable